includes/widgets/traits/button-trait.php

```diff
@@ -0,0 +1,564 @@
+	 */
+	protected function render_button( Widget_Base $instance = null ) {
+		if ( empty( $instance ) ) {
+			$instance = $this;
+		}
+
+		$settings = $instance->get_settings_for_display();
+
+		$instance->add_render_attribute( 'wrapper', 'class', 'elementor-button-wrapper' );
+
+		if ( ! empty( $settings['link']['url'] ) ) {
+			$instance->add_link_attributes( 'button', $settings['link'] );
+			$instance->add_render_attribute( 'button', 'class', 'elementor-button-link' );
+		}
+
+		$instance->add_render_attribute( 'button', 'class', 'elementor-button' );
+		$instance->add_render_attribute( 'button', 'role', 'button' );
+
+		if ( ! empty( $settings['button_css_id'] ) ) {
+			$instance->add_render_attribute( 'button', 'id', $settings['button_css_id'] );
+		}
+
+		if ( ! empty( $settings['size'] ) ) {
+			$instance->add_render_attribute( 'button', 'class', 'elementor-size-' . $settings['size'] );
+		}
+
+		if ( ! empty( $settings['hover_animation'] ) ) {
+			$instance->add_render_attribute( 'button', 'class', 'elementor-animation-' . $settings['hover_animation'] );
+		}
+		?>
+		<div <?php $instance->print_render_attribute_string( 'wrapper' ); ?>>
+			<a <?php $instance->print_render_attribute_string( 'button' ); ?>>
+				<?php $this->render_text( $instance ); ?>
+			</a>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Render button widget output in the editor.
+	 *
+	 * Written as a Backbone JavaScript template and used to generate the live preview.
+	 *
+	 * @since  3.4.0
+	 * @access protected
+	 */
+	protected function content_template() {
+		?>
+		<#
+		view.addRenderAttribute( 'text', 'class', 'elementor-button-text' );
+		view.addInlineEditingAttributes( 'text', 'none' );
+		var iconHTML = elementor.helpers.renderIcon( view, settings.selected_icon, { 'aria-hidden': true }, 'i' , 'object' ),
+		migrated = elementor.helpers.isIconMigrated( settings, 'selected_icon' );
+		#>
+		<div class="elementor-button-wrapper">
+			<a id="{{ settings.button_css_id }}" class="elementor-button elementor-size-{{ settings.size }} elementor-animation-{{ settings.hover_animation }}" href="{{ settings.link.url }}" role="button">
+				<span class="elementor-button-content-wrapper">
+					<# if ( settings.icon || settings.selected_icon ) { #>
+					<span class="elementor-button-icon elementor-align-icon-{{ settings.icon_align }}">
+						<# if ( ( migrated || ! settings.icon ) && iconHTML.rendered ) { #>
+							{{{ iconHTML.value }}}
+						<# } else { #>
+							<i class="{{ settings.icon }}" aria-hidden="true"></i>
+						<# } #>
+					</span>
+					<# } #>
+					<span {{{ view.getRenderAttributeString( 'text' ) }}}>{{{ settings.text }}}</span>
+				</span>
+			</a>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Render button text.
+	 *
+	 * Render button widget text.
+	 *
+	 * @param \Elementor\Widget_Base|null $instance
+	 *
+	 * @since  3.4.0
+	 * @access protected
+	 */
+	protected function render_text( Widget_Base $instance = null ) {
+		if ( ! $instance ) {
```

Comment:
@StyleShit can you explain why?

Reply:
It's the same as here:
https://github.com/elementor/elementor/blob/2094102d98b0d2735a853bdf7777e418a76cac8b/includes/widgets/traits/button-trait.php#L429-L432

Reply:
```suggestion
		// The default instance should be `$this` (a Button widget), unless the Trait is being used from outside of a widget (e.g. `Skin_Base`) which should pass an `$instance`.
		if ( empty( $instance ) ) {
```